examples/gno.land/p/demo/ujson/parse.gno

```diff
@@ -0,0 +1,569 @@
+package ujson
+
+import (
+	"std"
+	"strconv"
+	"strings"
+
+	"gno.land/p/demo/avl"
+)
+
+// https://stackoverflow.com/a/4150626
+const whitespaces = " \t\n\r"
+
+type FromJSONAble interface {
+	FromJSON(ast *JSONASTNode)
+}
```

Comment:
I don't like this personally. JSON is a simple format with well-established conventions in Go; the most known of these is that types which should do custom JSON En/decoding should implement `MarshalJSON` and `UnmarshalJSON`, with well-defined signatures. I'd prefer if we kept it that way, even while we don't have `encoding/json`, so that when we do have reflection-based json encoding/decoding, it is relatively painless to port the old code to the new one.

Reply:
Also, as a bonus, I also like `MarshalText` if you want to implement support for that :)

Reply:
I was trying to avoid using the same naming as `encoding/json` to prevent confusion